proxy/src/control/destination/background.rs

```diff
@@ -33,24 +33,53 @@ use control::{
     cache::{Cache, CacheChange, Exists},
     fully_qualified_authority::FullyQualifiedAuthority,
     remote_stream::{Receiver, Remote},
-    AddOrigin, Backoff, LogErrors
+        AddOrigin, Backoff, LogErrors
 };
 use dns::{self, IpAddrListFuture};
 use telemetry::metrics::DstLabels;
 use transport::{DnsNameAndPort, HostAndPort, LookupAddressAndConnect};
 use timeout::Timeout;
 use transport::tls;
 
-type DestinationServiceQuery<T> = Remote<PbUpdate, T>;
-type UpdateRx<T> = Receiver<PbUpdate, T>;
+pub trait GetDst {
+    type Error: fmt::Debug;
+    type Future: Future<
+        Item = grpc::Response<Self::Stream>,
+        Error = grpc::Error<Self::Error>,
+    >;
+    type Stream: Stream<Item = PbUpdate, Error = grpc::Error>;
+
+    fn poll_ready(&mut self) -> Poll<(), grpc::Error<Self::Error>>;
+
+    fn get_dst(&mut self, dst: Destination) -> Self::Future;
+}
+
+impl<T> GetDst for DestinationSvc<T>
+where
+    T: HttpService<RequestBody = tower_h2::BoxBody>,
+    T::ResponseBody: tower_h2::Body<Data = tower_h2::Data>,
+    T::Error: fmt::Debug,
+{
+    type Error = T::Error;
+    type Future = grpc::client::server_streaming::ResponseFuture<PbUpdate, T::Future>;
+    type Stream = grpc::Streaming<PbUpdate, T::ResponseBody>;
+
+    fn poll_ready(&mut self) -> Poll<(), grpc::Error<Self::Error>> {
+        DestinationSvc::poll_ready(self)
+    }
+
+    fn get_dst(&mut self, dst: Destination) -> Self::Future {
+        DestinationSvc::get(self, grpc::Request::new(dst))
```

Comment:
TIOLI: I think this could also just be `self.get(grpc::Request::new(dst)`, since (unlike `poll_ready` above) it should be unambiguous what method is being called?